calico_mesos/calico_mesos.py

```diff
@@ -493,8 +509,41 @@ def allocate(args):
         except TypeError:
             raise IsolatorException("num_ipv6 must be an integer")
 
-    return _allocate(num_ipv4, num_ipv6, hostname, uid)
-
+    # Check if the user has requested a specific IP via label
+    # Note: this will be deprecated once marathon provides the ipv4_addrs   
+    # field which will trigger netmodules' 'reserve'.
+    # This implementation replaces the requested IPAM IP with the static IP.
+    # i.e. If a user requests 1 IP, and specifies one specific IP in the 
+    # ipv4_addr label, they will receive 1 IP back - the one they specified.
+    ipv4_addrs = []
+    if labels and labels.has_key("ipv4_addrs"):
+        try:
+            ipv4_addrs = eval(labels['ipv4_addrs'])
+        except SyntaxError:
+            raise IsolatorException("Calico detected a malformed ipv4_addrs " \
+                    "field. Ensure you've specified a string representation " \
+                    "of a list of strings.")
+        if type(ipv4_addrs) != list:
```

Comment:
`reserve` will sanitize the IP addresses. We just need to make sure they're in a list as that's what `reserve` expects. I'll add a comment indicating that